copy/copy.go

```diff
@@ -241,11 +242,15 @@ func copyLayers(manifestUpdates *types.ManifestUpdateOptions, dest types.ImageDe
 		cl, ok := copiedLayers[srcLayer.Digest]
 		if !ok {
 			fmt.Fprintf(reportWriter, "Copying blob %s\n", srcLayer.Digest)
-			destInfo, diffID, err := copyLayer(dest, rawSource, srcLayer, diffIDsAreNeeded, canModifyManifest, reportWriter)
-			if err != nil {
-				return err
+			if dest.CopyForeignLayers() && len(srcLayer.URLs) != 0 {
+				destInfo, diffID, err := copyLayer(dest, rawSource, srcLayer, diffIDsAreNeeded, canModifyManifest, reportWriter)
+				if err != nil {
+					return err
+				}
+				cl = copiedLayer{blobInfo: destInfo, diffID: diffID}
+			} else {
+				fmt.Fprintf(reportWriter, "Skipping foreign layer copy to %s\n", dest.Reference().Transport().Name())
```

Comment:
We still need to fill in info for `manifestUpdates.{InformationOnly.,}LayerInfos` and possibly for `manifestUpdates.InformationOnly.LayerDiffIDs` which is much more horrible.

Also we might want to record URLs for the newly created layers as well.